Feature/import files merged

mbaertschi opened the pull request.

hwuethrich committed:
Allow updating column mappings

cloedu87 committed:
mapping of the dwc attributes of our demo file; modal with listing of our static demo mapping; demo-confirmation modal; live actions jump through modals; additional attributes for the records to complete basic demo file mapping and (later) upload; fix translations; fix tests

cloedu87 committed:
confirmation box

cloedu87 committed:
mapping importing values to column structs during import; changeset mutations for import via columns; new test-data; necessary data model changes for record resource; test

cloedu87 committed:
tests fixed

cloedu87 committed:
Merge remote-tracking branch 'origin/main' into feature/import-files

cloedu87 committed:
i18n

cloedu87 committed:
merge origin/main; testing; i18n; docs

cloedu87 committed:
Merge remote-tracking branch 'origin/main' into feature/import-files

cloedu87 committed:
Refactor ImportFile -> Import

hwuethrich committed:
Initial import creating records

hwuethrich committed:
Add routes for imports

cloedu87 committed:
live views for import detail and mapping overview WIP

cloedu87 committed:
i18n

cloedu87 committed:
remove dbg

cloedu87 committed:
import file process WIP; actions and basic content; navigation and icons

cloedu87 committed:
fix typo

cloedu87 committed:
fix typo

hwuethrich committed:
Merge branch 'main' into feature/import-files

* main:
  Add linters for gettext and migrations

hwuethrich committed:
Refactor imports

cloedu87 committed:
import record file user flow; navigation; stat card; i18n; tests

cloedu87 committed:
Merge remote-tracking branch 'origin/main' into feature/import-files

cloedu87 committed:
Merge branch 'feature/import-files' of https://github.com/zebbra/data_aggregator into feature/import-files

cloedu87 committed:
fix tests and linting

cloedu87 committed:
record import triggered from frontend; some demo dashboards; due to failing i18n -> reexport

mbaertschi committed:
Merge branch 'main' into feature/import-files-merged

hwuethrich commented on assets/js/hooks/viewport_resize.hook.ts:
@mbaertschi hmm, warum muss das backend die viewport size kennen?

hwuethrich commented on lib/data_aggregator_web.ex:
```diff
@@ -79,15 +79,31 @@ defmodule DataAggregatorWeb do
+      unquote(page_helpers())
+    end
+  end
+
```
@mbaertschi würde ich weglassen und einfach an den 2 stellen wo es verwendet wird halt das importen wo du brauchst (resp. `use ..., :html`).

hwuethrich commented on lib/data_aggregator_web.ex:
```diff
@@ -99,6 +115,21 @@ defmodule DataAggregatorWeb do
+      # Routes generation with the ~p sigil
+      unquote(verified_routes())
+    end
+  end
```
@mbaertschi was ist der unterschied zu `html_helpers`? kann man nicht alles da importen?

hwuethrich commented on lib/data_aggregator_web.ex:
```diff
@@ -79,15 +79,31 @@ defmodule DataAggregatorWeb do
       import DataAggregatorWeb.HeadlessComponents
       import DataAggregatorWeb.Headless.Dialog, only: [dialog_title: 1, dialog_description: 1]
       import DataAggregatorWeb.Gettext
+      import DataAggregatorWeb.Page
```
@mbaertschi DataAggregatorWeb.Page ist eine component, nicht? evtl. werden damit noch ein paar functions importiert die man evtl. nicht will

hwuethrich commented on lib/data_aggregator_web/viewport_helpers.ex:
Ich habe noch nicht genau gesehen warum man die viewport width braucht. Aber scheint mir etwas kompilizert... 😅 

Dieses Modul müsste/sollte kein Macro sein. Die helpers kann man dort importen wo man sie braucht. Und das `handle_event` könnte man in einem live hook machen, nicht?

hwuethrich commented on lib/data_aggregator_web/viewport_helpers.ex:
```diff
@@ -0,0 +1,123 @@
+  """
+  defmacro __using__(_) do
+    quote do
+      import DataAggregatorWeb.ViewportHelpers
```
Das macht momentan nix, da ViewportHelpers keine functions hat (alle sind im macro definiert).

hwuethrich committed:
Misc refactorings

* Create common context `Records` for `Record`, `Collection` and `Import`
* Use existing `assertions` library in tests
* Move API related modules to `DataAggregatorApi` context
* Use separate router for APIs (JSON/GraphQL) and forward `/api` in main router

hwuethrich committed:
Remove http_stream (not used atm)

hwuethrich committed:
Use assert_map_contains for clarity

hwuethrich committed:
Add calculation for import data

Also use Explorer.DataFrame.rename/2 to map column names for better performance.

hwuethrich committed:
Refactor digitizing_progress as calculation

cloedu87 committed:
small layout fixes and .env file catching for local dev usage; fixing gettext files

hwuethrich commented on config/runtime.exs:
```diff
@@ -12,6 +12,10 @@ if config_env() in [:dev, :test] do
   Envy.load(["config/.env.#{config_env()}"])
 end
 
+if config_env() in [:dev] do
+  Envy.load([".env"])
+end
+
```
@cloedu87 please use `direnv` for this. Using Envy in the last project was a semi-good idea... ;)

hwuethrich commented on lib/data_aggregator/records/collection.ex:
```diff
@@ -43,6 +43,14 @@ defmodule DataAggregator.Records.Collection do
     end
   end
 
+  calculations do
+    calculate :digitizing_progress,
+              :float,
+              expr(
+                if items_to_digitize > 0, do: 100 / (items_to_digitize * records_count), else: 0
+              )
+  end
```
@cloedu87 Ash is great for this! :D Not sure if the formula is right, thou

hwuethrich commented on lib/data_aggregator_web/live/collection_live/show.ex:
```diff
@@ -12,7 +12,7 @@ defmodule DataAggregatorWeb.CollectionLive.Show do
 
   @impl true
   def handle_params(%{"id" => id} = params, _url, socket) do
-    collection = DataAggregator.Records.load!(Collection.get_by_id!(id), [:records_count])
```
@cloedu87 all read actions accept a `load:` option directly

hwuethrich commented:
@mbaertschi @cloedu87 merged! hope this is ok but the PR is getting a bit big
